docs/RBAC.md

```diff
@@ -0,0 +1,152 @@
+
+![](https://static.swimlanes.io/2ee006a19f5dee69ece614e95f72d480.png)
+
+
+## Protect the Search API
+<!-- [NEW IN V2] This feature is new for V2. -->
+The API itself is protected by RBAC. Users must be given a role that allows access to search.
+
+The default ACM admin and viewer roles should include access to the Search API by default.
+
+> **OPEN DISCUSSION:**
+>
+> We are still investigating the implementation of this feature. We'll update as we learn more.
+>
+> Options:
+> 1. Kube API server extension - This can add additional load to the kube API server.
+> 2. Management Ingress - This is being deprecated.
+> 3. Validate access at the service - Provides the most flexibility, but how will it leverage kubernetes implementation?
+
+## Authenticate the user 
+
+The Search API authenticates the user (or service account) and impersonates the user to obtain their access rules.
+We cache the token validation for a short period (60 seconds) to reduce requests to the kube api. Tokens have a short expiration, so we must revalidate often. **Note that** this time-to-live period is independent of the cached rules, which is longer at 10 minutes of inactivity.
+
+> Use the [TokenReview API](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#tokenreview-v1-authentication-k8s-io) to validate the user token and obtain the UserInfo (username and groups).
+
+## Obtain the user's RBAC rules
+After authenticating the user, we obtain their authorization rules. There's 2 different scenarios:
+1. [Rules for resources in the hub cluster](#hub-cluster)
+2. [Rules for resources in managed clusters](#managed-clusters)
+
+> Use [User Impersonation](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#user-impersonation) when sending API requests on behalf of the end user.
+```
+CHANGE IN V2. [Does not affect API results.]
+V1 passes the user's token directly to the API, but impersonation is desired.
+This change doesn't change the semantic of the API.
+```
+
+### 1. Hub cluster
+
+The search results must include **exactly** the same resources users are authorized to list using kubectl, oc, or the kubernetes API on the OpenShift cluster hosting the ACM Hub.
+
+We request all the authorization rules for the user and [cache](#cache) the results.
+> 1. Get all resources in the cluster that support `list` and `watch`. This is shared across all users.
```

Comment:
Why do we need watch ? list isnt enough ?

Reply:
The collector only "collects" resources that support `watch` so if the resource doesn't support watch we ignore it because we know it's not in our data.
Similarly, if the resource doesn't support `list`, we know that users won't have permission to list because the operation isn't supported by the resource.